### DISCUSSION TITLE
Its Possible login using Formdata() on next-auth ?

### DISCUSSION THREAD

**Discussion by isggwp** [nextauthjs/next-auth#3148] [Help] [answered]
### Question 💬

Its Possible login using Formdata() on next-auth inside file `/pages/api/auth/[...nesxtauth.js]` ?

### How to reproduce ☕️

sorry its just simple question but mybe you can pull `next-auth-example` for reproduce :)

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by deadcoder0904** [comment]
Yes, it is totally possible to do that.

In fact, here's [the exact tutorial](https://spacejelly.dev/posts/how-to-make-twitter-api-requests-with-nextauth-js-session-tokens/#step-3-making-a-search-request-to-the-twitter-api-using-a-nextauthjs-authenticated-session) that does this, check out **Step 3: Making a search request to the Twitter API using a NextAuth.js authenticated session**. Here's the [1st part of that tutorial](https://spacejelly.dev/posts/how-to-authenticate-next-js-apps-with-twitter-nextauth-js/) :)

**Reply by balazsorban44** [reply]
The OP was curious to send `FormData` to the backend at login. Your link shows an example where the writer makes a POST request with JSON to their own API route.

**Reply by deadcoder0904** [reply]
Oops, makes sense 👍

**Comment by balazsorban44** [comment, marked as answer]
The `next-auth` signin method sends a `"Content-Type": "application/x-www-form-urlencoded"` header as shown here:

https://github.com/nextauthjs/next-auth/blob/ba39efb2566466dfe695bb90513050896d96a2d2/src/react/index.tsx#L211-L223

But since we are using Next.js and it does automatic body parsing, it might work, however, I haven't tried it, and most likely you will have to write your or `signIn` method.

I am curious what is your use case? Maybe we can find another way.

**Comment by chettriyuvraj** [comment]
The next-auth documentation actually provides a way to 'hijack' their built in token option and instead do your own thing (i.e make the post request however you want) and then return the token. 

This is to be done in the provider options parameter you specify in the [...nextauth.js] file.

![image](https://user-images.githubusercontent.com/32122172/207534189-7362e500-5557-4bfc-ac68-02470efa8150.png)

Doc link: https://next-auth.js.org/configuration/providers/oauth#how-to

**Reply by chettriyuvraj** [reply]
This same thing applies to other parts of the login as well (authorization, userInfo et al)
